src/sql-parser/src/parser.rs

```diff
@@ -3240,94 +3240,67 @@ impl<'a> Parser<'a> {
+                Ok(Statement::DropSchema(DropSchemaStatement {
                     name,
                     if_exists,
                     cascade,
-                }));
+                }))
             }
-            ROLE | USER => {
-                let if_exists = self.parse_if_exists()?;
+            ObjectType::Role => {
                 let names = self.parse_comma_separated(Parser::parse_object_name)?;
-                return Ok(Statement::DropRoles(DropRolesStatement {
+                Ok(Statement::DropRoles(DropRolesStatement {
                     if_exists,
                     names,
-                }));
+                }))
             }
-            CLUSTER => {
-                return if self.peek_keyword(REPLICA) {
-                    self.parse_drop_cluster_replicas()
-                } else {
-                    self.parse_drop_clusters()
-                };
+            ObjectType::Cluster => self.parse_drop_clusters(if_exists),
+            ObjectType::ClusterReplica => self.parse_drop_cluster_replicas(if_exists),
+            ObjectType::Table
+            | ObjectType::View
+            | ObjectType::MaterializedView
+            | ObjectType::Source
+            | ObjectType::Sink
+            | ObjectType::Index
+            | ObjectType::Type
+            | ObjectType::Secret
+            | ObjectType::Connection => {
+                let names = self.parse_comma_separated(Parser::parse_object_name)?;
+                let cascade = matches!(
+                    self.parse_at_most_one_keyword(&[CASCADE, RESTRICT], "DROP")?,
+                    Some(CASCADE),
+                );
+                Ok(Statement::DropObjects(DropObjectsStatement {
+                    object_type,
+                    if_exists,
+                    names,
+                    cascade,
+                }))
             }
-            INDEX => ObjectType::Index,
-            SINK => ObjectType::Sink,
-            SOURCE => ObjectType::Source,
-            TABLE => ObjectType::Table,
-            TYPE => ObjectType::Type,
-            VIEW => ObjectType::View,
-            MATERIALIZED => {
-                self.expect_keyword(VIEW)?;
-                ObjectType::MaterializedView
-            }
-            SECRET => ObjectType::Secret,
-            CONNECTION => ObjectType::Connection,
-            _ => unreachable!(),
-        };
-
-        let if_exists = self.parse_if_exists()?;
-        let names = self.parse_comma_separated(Parser::parse_object_name)?;
-        let cascade = matches!(
-            self.parse_at_most_one_keyword(&[CASCADE, RESTRICT], "DROP")?,
-            Some(CASCADE),
-        );
-        Ok(Statement::DropObjects(DropObjectsStatement {
-            object_type,
-            if_exists,
-            names,
-            cascade,
-        }))
+        }
     }
 
-    fn parse_drop_clusters(&mut self) -> Result<Statement<Raw>, ParserError> {
-        let if_exists = self.parse_if_exists()?;
```

Comment:
Nice! it looks like `CLUSTER` was the only case in which we didn't parse `if_exists` in the calling code, now we do, which is a nice win for unification!